http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -148,21 +149,23 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
           retrieveWsmDatabases(wsmResourceApi, referenceDatabaseNames, params.workspaceId.value)
         } else F.pure(List.empty)
 
-      // Create WSM kubernetes namespace
-      wsmNamespace <- childSpan("createWsmKubernetesNamespaceResource").use { implicit ev =>
-        createWsmKubernetesNamespaceResource(
-          app,
-          params.workspaceId,
-          namespacePrefix,
-          wsmDatabases.map(_.wsmDatabaseName),
-          wsmManagedIdentityOpt.map(_.wsmResourceName)
-        )
+      // Create or fetch WSM kubernetes namespace
+      wsmNamespace <- retrieveWsmNamespace(wsmResourceApi, namespacePrefix, params.workspaceId.value).flatMap {
```

Comment:
Thinking out loud, I think I like the approach of two API calls here instead of what we did for making google idempotent, which was catching 409s (in the case of create) and 404s (in the case of delete). Makes it a lot more explicit and resilient to external changes at the cost of an additional API call. 

Curious to get other thoughts here to make sure there's not other trade-offs I'm missing